src/main/java/org/jabref/gui/maintable/BibEntryTableViewModel.java

```diff
@@ -67,19 +69,15 @@ public BibEntry getEntry() {
         return entry;
     }
 
-    private static ObservableValue<List<AbstractGroup>> createMatchedGroupsBinding(BibDatabaseContext database, BibEntry entry) {
-        Optional<GroupTreeNode> root = database.getMetaData().getGroups();
-        if (root.isPresent()) {
-            return EasyBind.map(entry.getFieldBinding(StandardField.GROUPS), field -> {
-                List<AbstractGroup> groups = root.get().getMatchingGroups(entry)
-                                                 .stream()
-                                                 .map(GroupTreeNode::getGroup)
-                                                 .collect(Collectors.toList());
-                groups.remove(root.get().getGroup());
-                return groups;
-            });
-        }
-        return new SimpleObjectProperty<>(Collections.emptyList());
+    private static Binding<List<AbstractGroup>> createMatchedGroupsBinding(BibDatabaseContext database, BibEntry entry) {
+        return new UiThreadBinding<>(EasyBind.combine(entry.getFieldBinding(StandardField.GROUPS), database.getMetaData().groupsBinding(),
```

Comment:
Should this binding be modified to depend on any changes in the bibentry? #7325 depends on keyword field, instead of group field. Free text search depends on all fields. Should it be in a different PR? Personally I'd prefer if `createMatchedGroupsBinding` could be changed to return a `FilteredList`, but that would likely take me a while to implement...

Reply:
Yes, I think it's a good idea to listen to changes of the entry. Let's discuss this in a follow-up PR, since this one looks already good.